test/append.test.js

```diff
@@ -258,16 +258,88 @@ describe('SourceMap - Append Mappings', () => {
     assert.deepEqual(map.getMap(), expectedResultOne);
   });
 
-  it('Append buffer mappings with line and column offset', () => {
+  it('Append source map with line offset', () => {
     let map = new SourceMap('/test-root');
     map.addVLQMap({
       mappings: SIMPLE_SOURCE_MAP.mappings,
       sources: SIMPLE_SOURCE_MAP.sources,
       names: SIMPLE_SOURCE_MAP.names,
     });
-    let buffer = map.toBuffer();
-    map.addBuffer(buffer, 10, 46);
-    assert.deepEqual(map.getMap(), expectedResultTwo);
+    let map2 = new SourceMap(map.toBuffer());
+    map.addSourceMap(map2, 10);
+    assert.deepEqual(map.getMap(), expectedResultOne);
+    // map2 is consumed
+    assert.equal(map2.getMappings().length, 0);
+  });
+
+  it('Append source map with line offset that overlap original lines', () => {
```

Comment:
This is another case where this new implementation fails.

I guess we have to choose between accuracy and speed here.

Reply:
Hmm, when would two source maps overlap?

Reply:
Yeah probably never, as it would probably produce an invalid sourcemap regardless if they overlap